src/main/java/com/onepagecrm/models/serializers/EmailMessageSerializer.java

```diff
@@ -0,0 +1,110 @@
+package com.onepagecrm.models.serializers;
+
+import com.onepagecrm.exceptions.OnePageException;
+import com.onepagecrm.models.Email;
+import com.onepagecrm.models.EmailMessage;
+import com.onepagecrm.models.EmailRecipients;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class EmailMessageSerializer extends BaseSerializer {
+
+    private static final Logger LOG = Logger.getLogger(EmailMessageSerializer.class.getName());
+
+    public static List<EmailMessage> fromString(String responseBody) throws OnePageException {
+        List<EmailMessage> result = new ArrayList<>();
+        try {
+            String parsedResponse = (String) BaseSerializer.fromString(responseBody);
+            JSONObject responseObject = new JSONObject(parsedResponse);
+            return fromJsonArray(responseObject.getJSONArray("email_messages"));
+        } catch (ClassCastException e) {
+            throw (OnePageException) BaseSerializer.fromString(responseBody);
+        } catch (Exception e) {
+            LOG.severe("Error parsing Contact object from response body");
+            LOG.severe(e.toString());
+        }
+        return result;
+    }
+
+    public static List<EmailMessage> fromJsonArray(JSONArray emailsArray) {
+        List<EmailMessage> emails = new ArrayList<>();
+        for (int j = 0; j < emailsArray.length(); j++) {
+            JSONObject emailObject;
+            try {
+                emailObject = emailsArray.getJSONObject(j);
+                emails.add(fromJsonObject(emailObject));
+            } catch (JSONException e) {
+                LOG.severe("Error parsing email array");
+                LOG.severe(e.toString());
+            }
+        }
+        return emails;
+    }
+
+    public static EmailMessage fromJsonObject(JSONObject emailObject) {
+        EmailMessage email = new EmailMessage();
+        try {
+            emailObject = emailObject.getJSONObject("email_message");
```

Comment:
@fraggjkee - I always try to use `optJSONObject` instead of `getJSONObject` (same with String, int, Array etc.). The benefit is that, if any field is missing, the object will fully populated other than that field. Using `get...` will cause us to jump to the catch statement and miss the other fields as well. I know that APIs are not supposed to introduce breaking changes, but better to be able to handle them in any case in my own experience!

Reply:
Good point ;)